modules/standard/Socket.md

```diff
@@ -0,0 +1,210 @@
+# Socket Module API
+
+To those familiar with the Unix socket API, the method names will feel familiar, though their usage will be somewhat simpler than the raw Unix socket API. The Module will comprise of four records which will abstract lower order functions to higher and more intutive methods for user:
+- ipAddr : Address information storage structure.
+- tcpConn : A TCP Socket with established connection
+- tcpListener : A TCP Socket listening for connections
+- udpSocket : A connection less UDP Socket
+
+## Address Information
+
+`ipAddr` serves the purpose of storing address information in a single record irrespective of the protocol used which varies the buffer required to store address.
+
+To initialize an `ipAddr` there are several available overloads and defaults.
+
+**Method**
+```python
+proc ipAddr.init(host:string = "localhost", port:uint(16) = 8000, family:IPFamily = IPFamily.IPv4)
+```
+**Parameters**
+```
+host: string - address string in dot or color notation depending on family. "127.0.0.1" or "::1"
+port: uint(16) - port number for socket
+family: IPFamily - Internet Protocol version which can be IPFamily.IPv4 or IPFamily.IPv6
+```
+
+There are two more initialzers which can take in standard IP Addresses like loopback and broadcast as `host` which are defined as constants inside the `Sys` module.
+
+**Method**
+```python
+proc ipAddr.init(host:sys_in_addr_t, port: uint(16) = 8000)
+proc ipAddr.init(host:sys_in6_addr_t, port: uint(16) = 8000)
+```
+**Parameter**
+```
+host: string - standard ip addresses
+port: uint(16) - port number for socket
+```
+
+Family is determined based on type of standard address used `sys_in_addr_t` for ipv4 and `sys_in6_addr_t` for ipv6. Details about available options can be found [here](https://www.ibm.com/docs/en/zos/2.3.0?topic=considerations-special-ip-addresses#ipv6d0131000420__specialip)
```

Comment:
Probably better to reference https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/netinet_in.h.html here. Note that POSIX leaves the IPv4 loopback address as implementation-defined, though it says most systems do define it, as `INADDR_LOOPBACK`.

Reply:
making the change thanks 👍🏻

Reply:
regarding `INADDR_LOOPBACK` will the user and the code too need `ifdef` checks or we can assume that `INADDR_LOOPBACK` will be available on majority of systems ?

Reply:
We can define it for our own use if it isn't defined (but such things have to happen in C code).